portability-spi-transfer/src/main/java/org/datatransferproject/spi/transfer/provider/SignalHandler.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2024 The Data Transfer Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.datatransferproject.spi.transfer.provider;
+
+import java.io.IOException;
+import java.util.UUID;
+import org.datatransferproject.api.launcher.Monitor;
+import org.datatransferproject.spi.transfer.types.CopyExceptionWithFailureReason;
+import org.datatransferproject.spi.transfer.types.signals.SignalType;
+import org.datatransferproject.types.transfer.auth.AuthData;
+import org.datatransferproject.types.transfer.auth.TokensAndUrlAuthData;
+import org.datatransferproject.types.transfer.retry.RetryException;
+
+/**
+ * Signals the status of the Job to the corresponding Service.
+ */
+public interface SignalHandler<A extends AuthData> {
+  default void sendSignal(SignalRequest signalRequest, AuthData authData, Monitor monitor)
```

Comment:
Maybe a bit lost in my earlier comment, but wanted to reiterate that `authData` feels out of place here - that you need `AuthData` to send the signal is an implementation detail of the `AppleSignalHandler` which is leaking through to the abstraction here.

Reply:
If a provider makes a decision to implement their SignalHandler then that would have to be a public API protected by some Auth Material. This auth material should guarantee that worker is updating the job that belongs to its owner. The default implementation just ignores it.

Reply:
I agree that if you're calling an API with your `SignalHandler` it should be authorized, what I'm suggesting is that passing the auth data in the `sendSignal` call leaks that implementation detail - that the default implementation ignores it suggests it's better provided elsewhere, for example in the constructor of specific `SignalHandler`s that require it

Reply:
Per our conversation during the maintainers meeting, resolving as it's consistent with existing APIs